src/locales.js

```diff
@@ -11,54 +11,53 @@ module.exports = {
      * Pull a translation file
      * @param  {Array}   locales Locales to pull ( by default pulls all of them )
      */
-    pull: function(locales = []){
+    pull: async function(locales = []){
+        let files;
+        try {
+            files = await this.getAllFiles();
+        } catch (err) {
+            console.log(err);
+        }
+
+        let masterId = files.filter( f => f.master_project_file_id === null )[0].id;
+
+        //Filter locales to pull by the locales passed
+        if(locales.length){
+            files = files.filter( f => locales.indexOf(f.locale_code) > -1 );
         
-        this.getAllFiles()
-        .then( (files) => {
-            let masterId = files.filter( f => f.master_project_file_id === null )[0].id;
-
-            //Filter locales to pull by the locales passed
-            if(locales.length){
-                files = files.filter( f => locales.indexOf(f.locale_code) > -1 );
-            
-            //If no locales passed we pull all locales but the master one
-            } else {
-                files = files.filter( f => f.id !== masterId );
+        //If no locales passed we pull all locales but the master one
+        } else {
+            files = files.filter( f => f.id !== masterId );
+        }
+
+        let spinners = {};
+
+        // Download the content for each file
+        for(let fileIndex = 0; fileIndex < files.length; fileIndex++) {
+            const file = files[fileIndex];
+            spinners[file.locale_code] = Utils.createSpinner(`Pulling file for ${file.locale_code} locale in ${file.name}`);
+
+            let result;
+            try {
+                result = await Utils.get(`/files/${masterId}/locales/${file.locale_code}`);
+            } catch (err) {
+                console.log(err);
```

Comment:
Again if we don't do anything for the loop it will try to continue, if you catch the error. Maybe should also use the `Utils.handleError` for this.

Try to make it brake to test it as well.